### PR TITLE
[go-gen] Add main.go request struct generation

### PR DESCRIPTION
**Pull request by willjvsmith**
* Adds generation of request struct generation in the non-auth service main files
  * Adds placeholder parameter validation, ready for @jaylees14 
  * Adds temporary checks in `generateImports` to make the integration tests pass
* Updates unit, e2e and integration tests

This stuff gets quite tricky, but I think once these structs are done it should get a bit easier

### CODE REVIEW

File: src/main/scala/temple/generate/server/go/service/GoServiceMainGenerator.scala [hunk 1]
```diff
@@ -21,12 +27,28 @@ object GoServiceMainGenerator {
...
         //doubleQuote("net/http"),
+
+        // TODO: This check is temporary to make the integration tests pass
+        when(
+          clientAttributes
+            .exists {
+              case (_, attr) =>
+                attr.attributeType == AttributeType.DateType ||
+                attr.attributeType == AttributeType.TimeType ||
+                attr.attributeType == AttributeType.DateTimeType
```
Comment (c-harding) [inline src/main/scala/temple/generate/server/go/service/GoServiceMainGenerator.scala:38]: personally it feels like `Seq(AttributeType.DateType,AttributeType.TimeType,AttributeType.DateTimeType) contains attr.attributeType` would be nicer, but happy to be overruled

**Review by jaylees14 — APPROVED** [review]
LGTM 👍

File: src/main/scala/temple/generate/server/go/service/GoServiceMainGenerator.scala [hunk 2]
```diff
@@ -21,12 +27,28 @@ object GoServiceMainGenerator {
         //doubleQuote("fmt"),
         //doubleQuote("log"),
         //doubleQuote("net/http"),
+
+        // TODO: This check is temporary to make the integration tests pass
+        when(
+          clientAttributes
+            .exists {
+              case (_, attr) =>
```
Comment (c-harding) [inline src/main/scala/temple/generate/server/go/service/GoServiceMainGenerator.scala:35]: Likewise `clientAttributes.valuesIterator.exists { attr =>` feels nicer to me

**Review by c-harding — APPROVED** [review]
Looks nice!

File: src/main/scala/temple/generate/server/go/service/GoServiceMainGenerator.scala (hunk 1 above)
Comment (willjvsmith) [reply src/main/scala/temple/generate/server/go/service/GoServiceMainGenerator.scala:38]: temporary for integration tests to pass, see TODO comment in the lines above

**Pull request merged**